File: backends/php-laravel/app/UseCases/Auth/Dtos/SocialDto.php
<?php
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
namespace App\UseCases\Auth\Dtos;

use App\Domain\Staff\Enums\Provider;
use App\Support\Dtos\AbstractDto;

/**
 * ソーシャルDTOクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 * @package App\UseCases\Auth\Dtos
 */
class SocialDto extends AbstractDto
{
    public ?Provider $provider = null;
    public ?string $providerId = null;
    public ?string $nickname = null;
    public ?string $name = null;
    public ?string $email = null;
    public ?string $avatar = null;
}

File: backends/java-spring/src/main/java/com/authorization/usecases/auth/dtos/SocialDto.java
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
package com.authorization.usecases.auth.dtos;

import com.authorization.domain.staff.enums.Provider;
import com.authorization.support.dtos.AbstractDto;
import lombok.Getter;
import lombok.Setter;

/**
 * ソーシャルDTOクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
@Getter
@Setter
public class SocialDto extends AbstractDto {

    private Provider provider;
    private String providerId;
    private String nickname;
    private String name;
    private String email;
    private String avatar;
    private String invitationToken;
}
